CMakeLists.txt

```diff
@@ -111,6 +114,11 @@ find_package(Doxygen 1.8.5)
 
 find_package(GTest 1.6.0)
 
+if (USE_EXTERNAL_MFEM)
+    find_library(MFEM mfem  "${MFEM_DIR}/lib")
+    find_path(MFEM_INCLUDES mfem.hpp "${MFEM_DIR}/include")
+endif()
+
 if (USE_MFEM)
     find_library(MFEM mfem "${CMAKE_SOURCE_DIR}/dependencies/mfem" "${MFEM_DIR}/lib")
```

Comment:
I think I get it, but I am not sure if that will work. The external MFEM installation will also bring Hypre, Metis etc. So we do not have to import them again here. I propose that we use a construction such as.
```
if (USE_EXTERNAL_MFEM)
   ...
elseif(USE_MFEM)
   ...
endif()
```
I will pushed a commit with such a proposition. It is closely related to the comment above I think.

 I wonder if we should put in some checks to verify that MFEM indeed brings Hypre, Metis etc for the external mfem option. What do you think?

Reply:
> The external MFEM installation will also bring Hypre, Metis etc. So we do not have to import them again here. I propose that we use a construction such as.
> 
> ```
> if (USE_EXTERNAL_MFEM)
>    ...
> elseif(USE_MFEM)
>    ...
> endif()
> ```

I'm a little unsure about this. If this was the case, we wouldn't have had to include mfem dependencies from the beginning. In fact, in order to use the external mfem correctly, I think we do have to import all the dependencies explicitly as well.

In any cases, to me the suggestion I made above seems sufficient. At the point `MFEM_DIR` is specified and `cmake` does find the mfem from that path, it is clear that the installation is using external mfem. We could probably do the same path order change for other dependencies as well.

Reply:
Hi @dreamer2368 , a lot of stuff came in between, I am sorry for that. I just to a look at this again.

I think the current solution is the way to go for the following reasoning.

We have two ways of importing MFEM:
1. MFEM compiled/installed in the libROM folder (default)
2. An external installation with the location installed in  `${MFEM_DIR}/lib`.

The first and the second option might have different dependencies. They do not have to be different, but they might be different. Also, most probable they are installed in different locations, but this does not have to be. We find them in the following way for both options:
1. We find the dependencies installed specifically for the MFEM in the libROM folder by means of a search for these libraries independently.
2. We import the dependencies whilst importing the external MFEM. The cmake of MFEM is configured as such that uppon importing MFEM, it will also import its dependencies.

So in the second option, if we import the external MFEM **and** we import the dependencies, we might load the dependencies twice. The current set up avoids this.

I think the logic could be different in `CMakeLists.txt` but it would not give a different result.

@dreamer2368 Do you agree?

Remark 1: I am very unsure about this, but I would not be surprised if the separate finding of Hypre and (Par)Metis is not necessary for option 1. However, it would probably not hurt.

Remark 2: The way that importing MFEM will also bring its dependencies has not been implemented by me in this PR. For the current install, upon importing libROM, the dependencies have to be imported separately.

Reply:
Hi @dreamer2368 , I had a look again and I think you were right. I have implemented your suggestion and it works out! I left the comment above for documentation of the conversation, but you can ignore it.